integration-tests/test_motd.py

```diff
@@ -0,0 +1,72 @@
+import contextlib
+import os
+
+import pytest
+
+
+DOT_REGISTERED_PATH = "/etc/insights-client/.registered"
+DOT_UNREGISTERED_PATH = "/etc/insights-client/.unregistered"
+
+MOTD_PATH = "/etc/motd.d/insights-client"
+MOTD_SRC = "/etc/insights-client/insights-client.motd"
+
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+@pytest.fixture
+def delete_special_files():
+    """Clean special files before and after test.
+
+    Since .registered and .unregistered files are usually ignored and
+    do not affect the behavior of insights-client, the MOTD case is
+    special and depends on them.
+
+    To ensure the tests here are not affected by them, we remove them
+    """
+
+    def delete_files():
+        with contextlib.suppress(FileNotFoundError):
+            os.remove(DOT_REGISTERED_PATH)
+        with contextlib.suppress(FileNotFoundError):
+            os.remove(DOT_UNREGISTERED_PATH)
+        with contextlib.suppress(FileNotFoundError):
+            os.remove(MOTD_PATH)
+
+    delete_files()
+    yield
+    delete_files()
+
+
+@pytest.mark.usefixtures("delete_special_files")
+def test_motd(insights_client):
+    """MOTD only exists on unregistered system without (.un)registered files."""
+    # If the system is not registered, the file should be present.
+    insights_client.run("--status")
+    assert os.path.exists(MOTD_PATH)
+
+    # After registration, the file should not exist.
+    insights_client.register()
+    assert not os.path.exists(MOTD_PATH)
```

Comment:
IMO this is not the right way to test it, forcefully deleting motd file and then performing registration does not mimic the right test scenario.
test case could be after registration the  motd file should be deleted but if we already deleted manually what are we testing

Reply:
my suggestion is to not delete motd in delete_special_files